geo/src/algorithm/projection_1d.rs

```diff
@@ -0,0 +1,285 @@
+use crate::CoordFloat;
+use crate::geometry::*;
+
+/// Project polygon geometry from 2D space into 1D space.
+///
+/// This trait provides methods to collapse polygon points onto a single axis,
+/// while preserving references to the original 2D coordinates. This enables
+/// spatial indexing techniques where the 1D projection is used to identify
+/// candidate closest points, then true euclidean distances are calculated
+/// using the original coordinates.
+pub(crate) trait Project1D<T>
+where
+    T: CoordFloat,
+{
+    /// Project all polygon points onto the X-axis.
+    ///
+    /// Returns a vector of tuples where:
+    /// - First element: x-coordinate (projected value)
+    /// - Second element: index into the original polygon's vertex array
+    #[allow(dead_code)]
+    fn project_x(&self) -> Vec<(T, usize)>;
+
+    /// Project all polygon points onto the Y-axis.
+    ///
+    /// Returns a vector of tuples where:
+    /// - First element: y-coordinate (projected value)
+    /// - Second element: index into the original polygon's vertex array
+    #[allow(dead_code)]
+    fn project_y(&self) -> Vec<(T, usize)>;
+}
+
+impl<T> Project1D<T> for Polygon<T>
```

Comment:
Low priority, non-blocking: Rather than having separate implementations for `Polygon` and `MultiPolygon`, could it be a generic implementation on `G` where `G: CoordsIter`? https://docs.rs/geo/latest/geo/algorithm/coords_iter/trait.CoordsIter.html